src/hist.rs

```diff
@@ -0,0 +1,103 @@
+extern crate telemetry;
+
+use self::telemetry::plain::*;
+use std::collections::HashMap;
+
+pub struct Histograms {
+    #[allow(dead_code)] // Issue #33 -- Will go away with subsequent commits.
+    ts: telemetry::Service,
+    hist_storage: HashMap<String, telemetry::plain::Linear<u32>>
+}
+
+impl Histograms {
+    pub fn new() -> Histograms {
+        Histograms {
+            ts: telemetry::Service::new(true /* activate immediately */),
+            hist_storage: HashMap::new()
+        }
+    }
+
+    #[allow(dead_code)] // Issue #33 will go away with subsequent commits.
+    pub fn create_linear_histogram(&mut self, name: &str, min: u32, max: u32, buckets: usize) -> bool  {
+        let hist:telemetry::plain::Linear<u32> = telemetry::plain::Linear::new(
+                    &self.ts,
+                    name.to_string(),
+                    min,
+                    max,
+                    buckets);
+        let ret = self.hist_storage.insert(name.to_string(), hist);
```

Comment:
Do we need to worry about name collision here? I guess each train will have its own instance so there shouldn't be any name collision between trains. And I would think within a train they wouldn't use the same name for different histograms.